pkg/controllers/rolloutrun/executor/batch.go

```diff
@@ -192,19 +192,24 @@ func (e *batchExecutor) doBatchUpgrading(ctx *ExecutorContext) (bool, time.Durat
 	batchTargetStatuses := make([]rolloutv1alpha1.RolloutWorkloadStatus, 0)
 	workloadChanged := false
 	for _, item := range currentBatch.Targets {
-		wi := ctx.Workloads.Get(item.Cluster, item.Name)
-		if wi == nil {
+		info := ctx.Workloads.Get(item.Cluster, item.Name)
+		if info == nil {
 			return false, retryStop, newWorkloadNotFoundError(item.CrossClusterObjectNameReference)
 		}
 
+		batchTargetStatuses = append(batchTargetStatuses, info.APIStatus())
+
+		// if target is already updated, skip upgrading
+		if e.isTargetReady(info, item) {
+			continue
```

Comment:
I suggest not to add this check, because there exists race conditions when you upgrade workload and trigger rollout at the same time. Rollout may think the workload is already `Ready`, but the workload is not `Started` yet.

Reply:
make sense